net/httpbody.go

```diff
@@ -0,0 +1,270 @@
+package net
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"sync"
+
+	"github.com/zalando/skipper/metrics"
+)
+
+var (
+	ErrClosed  = errors.New("reader closed")
+	ErrBlocked = errors.New("blocked string match found in body")
+)
+
+const (
+	defaultReadBufferSize uint64 = 8192
+)
+
+type MaxBufferHandling int
+
+const (
+	MaxBufferBestEffort MaxBufferHandling = iota
+	MaxBufferAbort
+)
+
+type logBody struct {
+	ctx    context.Context
+	fmtstr string
+	log    func(format string, args ...interface{})
+	input  io.ReadCloser
+}
+
+func newLogBody(ctx context.Context, fmtstr string, log func(format string, args ...interface{}), rc io.ReadCloser) io.ReadCloser {
+	return &logBody{
+		ctx:    ctx,
+		fmtstr: fmtstr,
+		input:  rc,
+		log:    log,
+	}
+}
+
+func (lb *logBody) Read(p []byte) (int, error) {
+	n, err := lb.input.Read(p)
+	if n > 0 {
+		lb.log("%s%s", lb.fmtstr, p)
+	}
+	return n, err
+}
+
+func (lb *logBody) Close() error {
+	return lb.input.Close()
+}
+
+func LogBody(ctx context.Context, fmtstr string, log func(format string, args ...interface{}), rc io.ReadCloser) io.ReadCloser {
+	return newLogBody(ctx, fmtstr, log, rc)
+}
```

Comment:
It looks like a simple Reader wrapper. Why do we need it in net package?

Reply:
It also does not look related to WrapBody so maybe add it in a separate PR?

Reply:
It was your suggestion to add it separately from WrapBody